GoogleForCreators/web-stories-wp
# Workspace: Footer button updates and related refactoring

## Context

The buttons in the footer of the editor were to be updated with a 'badge' style used for the notification count. The Layers button also needed a disclosure (chevron) icon added to indicate to users that this was a menu-type UI element.

## Summary

Styling for the buttons was updated, disclosure icon added to the Layers button, and carousel width was adjusted to prevent interference with wider buttons.

## Relevant Technical Choices

- Footer toggle button component was refactored to more robustly handle child components/elements/text
- Style rules for icons were moved to the toggle button component since they shared the same styling
- `hasMenuList` prop was added to the `ToggleButton` component for conditional rendering of the chevron disclosure icon
- Styling of the Zoom button was updated to match the other footer buttons (it uses a different core component with different styling)
- Adding the 'badge' styling and disclosure icon to the Layers button increased the widths of the left and right areas in the footer, so the carousel grid sizing was adjusted to accommodate this. This resulted in a narrower 'window' to show page thumbnails, mainly affecting stories with a large number of pages and narrower browser viewports.
- General cleanup of sizing and spacing in footer button contents

## To-do

- Consider different layout of footer buttons to allow a wider carousel view.
- The `Disclosure` and `Badge` components should be extracted for re-use.
- Hard-coded values may need to be updated with constants defined in the design system.

## User-facing changes

Before (1440px viewport)
<img width="1030" alt="Screen Shot 2022-06-17 at 12 54 45 PM" src="https://user-images.githubusercontent.com/4173034/174352826-3a951d3d-be91-4e3e-bb9c-7ad5aafd28cc.png">

After (1440px viewport)
<img width="1028" alt="Screen Shot 2022-06-17 at 12 57 48 PM" src="https://user-images.githubusercontent.com/4173034/174352857-ef64f137-5f28-42b9-a941-2ceec137c6b4.png">

After (1920px viewport)
<img width="1514" alt="Screen Shot 2022-06-17 at 1 11 38 PM" src="https://user-images.githubusercontent.com/4173034/174354613-df7dda5b-88be-4067-a01c-c08da3214234.png">


## Testing Instructions

- Ensure functionality of footer buttons is retained
- Ensure consistent styling of footer buttons
- Check other `ToggleButton` component instances in the UI to ensure proper styling




## Reviews

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist




---



Fixes #10891 
Related to #8086 (review suggested)
Related to #11631 (review suggested)
